libraries/shared/factories/storage/transformer.go

```diff
@@ -38,14 +47,19 @@ func (transformer Transformer) GetContractAddress() common.Address {
 	return transformer.Address
 }
 
-func (transformer Transformer) NewTransformer(db *postgres.DB) transformer.StorageTransformer {
+func (transformer Transformer) NewTransformer(db *postgres.DB) ITransformer {
 	transformer.StorageKeysLookup.SetDB(db)
 	transformer.Repository.SetDB(db)
 	return transformer
 }
 
 func (transformer Transformer) KeccakContractAddress() common.Hash {
-	return types.HexToKeccak256Hash(transformer.Address.Hex())
+	emptyHash := common.Hash{}
+	if transformer.hashedAddress == emptyHash {
+		return types.HexToKeccak256Hash(transformer.Address.Hex())
```

Comment:
I think I'm missing where `hashedAddress` is getting set 🤔 

Is it meant to happen on this line?

Reply:
🤦‍♀ good catch! i'll push a fix up shortly.